### PR TITLE
Full contact repo review

### PR DESCRIPTION
**Pull request by cavis**

### CODE REVIEW

File: bin/application [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+
+PROCESS=
+CMD_ARGS=
+
+ApplicationUsage () {
+  echo "usage: application [ help | test | web | mix | compile | migrate | -- ]"
```
Comment (cavis) [inline bin/application]: Would be useful to also have a `console` in here.

File: priv/repo/migrations/20170815191331_update_shows_table.exs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+defmodule Jingle.Repo.Migrations.UpdateShowsTable do
+  use Ecto.Migration
+
+  def change do
+    alter table(:shows) do
+      add :feeder_podcast_id, :integer
+      add :jinglerk_site_id, :integer
```
Comment (cavis) [inline priv/repo/migrations/20170815191331_update_shows_table.exs]: `jinglerk`?  (I read it as "jerk_site_id").

File: priv/repo/migrations/20170815191331_update_shows_table.exs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+defmodule Jingle.Repo.Migrations.UpdateShowsTable do
+  use Ecto.Migration
+
+  def change do
+    alter table(:shows) do
```
Comment (cavis) [inline priv/repo/migrations/20170815191331_update_shows_table.exs:5]: This is also going to need the `prx_account_ids` array, to actually start using the auth token.  But yeah, probably still too early to be worrying about where that comes from.

File: test/controllers/api/campaign_controller_test.exs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  test "creates and renders resource when data is valid", %{conn: conn} do
+    conn = post conn, api_campaign_path(conn, :create), campaign: @valid_attrs
+    assert json_response(conn, 201)["id"]
+    assert Repo.get_by(Campaign, @valid_attrs)
+  end
+
+  test "does not create resource and renders errors when data is invalid", %{conn: conn} do
+    conn = post conn, api_campaign_path(conn, :create), campaign: @invalid_attrs
+    assert json_response(conn, 422)["errors"] != %{}
```
Comment (cavis) [inline test/controllers/api/campaign_controller_test.exs:43]: I _think_ the rest of our APIs go with 400 here.  But I'm pretty sure any client should be able to handle them the same.

File: web/models/api/campaign.ex [hunk 5]
```diff
@@ -0,0 +1,24 @@
+defmodule Jingle.API.Campaign do
```
Comment (cavis) [inline web/models/api/campaign.ex:1]: Why `/models/api` and not just `/models/`?  I'm not sure modeling after the view/controller routes makes sense here.

File: web/models/api/show.ex [hunk 6]
```diff
@@ -0,0 +1,24 @@
+defmodule Jingle.API.Show do
```
Comment (cavis) [inline web/models/api/show.ex:1]: I feel like we have too many nouns out there already that mean this.  (Program, podcast, series).  And "show" reminds me of "action = :show".

I have a lot of feelings.

File: web/router.ex [hunk 7]
```diff
@@ -13,27 +13,36 @@ defmodule Adze.Router do
     plug :accepts, ["json", "hal"]
   end
 
+  def id_host, do: Env.get(:id_host)
+  pipeline :authorized do
+    plug PrxAuth.Plug, required: true, iss: &Jingle.Router.id_host/0
+  end
+
   # pipeline :authorized do
```
Comment (cavis) [inline web/router.ex]: Leftover comment?

File: web/views/api/campaign_view.ex [hunk 8]
```diff
@@ -0,0 +1,43 @@
+defmodule Jingle.API.CampaignView do
+  use Jingle.Web, :view
+
+  def render("index.json", %{conn: conn, campaigns: campaigns}) do
+    %{
+      count: length(campaigns),
+      total: length(campaigns),
+      _embedded: %{
```
Comment (cavis) [inline web/views/api/campaign_view.ex:8]: Someday we should abstract some of this HAL-generation logic, so we can do things like:

```elixir
resp(length(campaigns))
|> embed("prx:items", the_campaigns_json)
|> link("self", %{"http://what.ever"})
```

But it is not this day!

File: web/views/api/root_view.ex [hunk 9]
```diff
@@ -1,65 +1,60 @@
...
-      # }
+      }
+    }
   end
 
-  # defp fix_path(path) do
-  #   path |> String.replace("%7B", "{") |> String.replace("%7D", "}")
-  # end
+  defp fix_path(path) do
+    path |> String.replace("%7B", "{") |> String.replace("%7D", "}")
```
Comment (cavis) [inline web/views/api/root_view.ex:57]: This looks familiar!

File: web/views/changeset_view.ex [hunk 10]
```diff
@@ -0,0 +1,19 @@
+defmodule Jingle.ChangesetView do
```
Comment (cavis) [inline web/views/changeset_view.ex:1]: Hrmm... not sure i understand what this is.  But I'll persevere and keep reading!

File: web/views/error_helpers.ex [hunk 11]
```diff
@@ -32,9 +32,9 @@ defmodule Adze.ErrorHelpers do
     #     dgettext "errors", "is invalid"
     #
     if count = opts[:count] do
-      Gettext.dngettext(Adze.Gettext, "errors", msg, msg, count, opts)
+      Gettext.dngettext(Jingle.Gettext, "errors", msg, msg, count, opts)
     else
-      Gettext.dgettext(Adze.Gettext, "errors", msg, opts)
+      Gettext.dgettext(Jingle.Gettext, "errors", msg, opts)
```
Comment (cavis) [inline web/views/error_helpers.ex:37]: Internationalization?  Fancy!

**Review by cavis — COMMENTED** [review]
Haven't actually used Ecto outside of an example app, so you may have to explain some stuff to me.  But generally, I think this approach makes sense.  The adzerk-API lib will likely be more complex, since you'll have to build it from scratch.  Sounds like fun.

Now, let the question-answering begin!

File: priv/repo/migrations/20170815191331_update_shows_table.exs (hunk 2 above)
Comment (gcampo88) [reply priv/repo/migrations/20170815191331_update_shows_table.exs]: ha! this was `adzerk` and i guess suffered from the find `adze` and replace

File: web/views/api/root_view.ex (hunk 9 above)
Comment (gcampo88) [reply web/views/api/root_view.ex:57]: does it look like it was lifted verbatim from castle? :no_mouth:

File: bin/application (hunk 1 above)
Comment (gcampo88) [reply bin/application]: 👍

File: priv/repo/migrations/20170815191331_update_shows_table.exs (hunk 3 above)
Comment (gcampo88) [reply priv/repo/migrations/20170815191331_update_shows_table.exs:5]: gotcha. will leave that for auth token implementation down the line!

File: web/router.ex (hunk 7 above)
Comment (gcampo88) [reply web/router.ex]: oops!

File: web/views/api/campaign_view.ex (hunk 8 above)
Comment (gcampo88) [reply web/views/api/campaign_view.ex:8]: 👍  ⚔️

File: web/views/changeset_view.ex (hunk 10 above)
Comment (gcampo88) [reply web/views/changeset_view.ex:1]: pretty sure phoenix made this all by itself

File: web/views/error_helpers.ex (hunk 11 above)
Comment (gcampo88) [reply web/views/error_helpers.ex:37]: this one too

File: web/models/api/show.ex (hunk 6 above)
Comment (gcampo88) [reply web/models/api/show.ex:1]: made a ticket! https://github.com/PRX/jingle.prx.org/issues/49

File: web/models/api/campaign.ex (hunk 5 above)
Comment (gcampo88) [reply web/models/api/campaign.ex:1]: made a ticket https://github.com/PRX/jingle.prx.org/issues/50

File: test/controllers/api/campaign_controller_test.exs (hunk 4 above)
Comment (gcampo88) [reply test/controllers/api/campaign_controller_test.exs:43]: made a ticket https://github.com/PRX/jingle.prx.org/issues/51

**Pull request closed without merging**